### PR TITLE
Fix Az Func Build and Docker Image

### PR DESCRIPTION
**Pull request by wsugarman**
## Description
- Use latest 3.1.x .NET Core SDK for both SQL code-gen and Az Func tooling
- Use .NET 5.0 SDK for Az Func docker build
- Fix target framework specification
- Remove global.json for Az Func project

## Related issues
N/A

## Testing
Pipelines passes and docker compose runs locally

### CODE REVIEW

File: docker/functions/Dockerfile [hunk 1]
```diff
@@ -9,8 +10,12 @@ ENV AzureWebJobsScriptRoot=/home/site/wwwroot \
     AzureFunctionsJobHost__Logging__Console__IsEnabled=true
 
 # Copy the DICOM Server repository and build the Azure Functions project
-FROM mcr.microsoft.com/dotnet/sdk:3.1.409-alpine3.12@sha256:3647a1c8c0b1c5533a3af9861e79299e07be33340f19b310ab9b74e2555e1b59 AS build
+# Note: While the latest .NET SDK is used for building, Azure Functions build target requires .NET Core 3.1 to be installed
+FROM mcr.microsoft.com/dotnet/sdk:5.0.301-alpine3.12@sha256:2e36595afe0c5e1f8c595ae0586d1c003817561afc8ab93586c6afde1d32e77c AS build
```
Comment (yuedai0531) [inline docker/functions/Dockerfile:14]: is this hash supposed to be here?

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -22,15 +22,16 @@ jobs:
     - name: Checkout repository
       uses: actions/checkout@v2.3.4
 
-    - name: Setup dotnet (for SQL generation)
+    - name: Setup dotnet from global.json
```
Comment (yuedai0531) [inline .github/workflows/codeql-analysis.yml:25]: I thought you removed `global.json`?

Comment (wsugarman) [reply .github/workflows/codeql-analysis.yml:25]: There were previously two. We keep a global JSON at the root to specify the .NET SDK used to build the solution. But I removed the one I previously added that was in the *Microsoft.Health.Dicom.Functions* folder. I initially imaged we'd use a separate SDK for building the functions, but instead we'll just keep the .NET 5.0 SDK for the solution overall.

File: Microsoft.Health.Dicom.sln [hunk 3]
```diff
@@ -41,12 +41,12 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.Health.Dicom.Clie
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.Health.Dicom.Functions", "src\Microsoft.Health.Dicom.Functions\Microsoft.Health.Dicom.Functions.csproj", "{BB88616C-6208-43D0-B9EF-79FC0652A151}"
 EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.Health.Dicom.Functions.UnitTests", "src\Microsoft.Health.Dicom.Functions.UnitTests\Microsoft.Health.Dicom.Functions.UnitTests.csproj", "{3D679950-A578-45AD-AF89-FAF89580375F}"
```
Comment (pengchen0692) [inline Microsoft.Health.Dicom.sln:44]: Do we need this change?

File: src/Microsoft.Health.Dicom.Functions/Microsoft.Health.Dicom.Functions.csproj [hunk 4]
```diff
@@ -2,20 +2,19 @@
 
   <PropertyGroup>
     <AzureFunctionsVersion>v3</AzureFunctionsVersion>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <TargetFrameworks>netcoreapp3.1</TargetFrameworks>
   </PropertyGroup>
 
+  <ItemGroup>
+    <FrameworkReference Include="Microsoft.AspNetCore.App" />
```
Comment (pengchen0692) [inline src/Microsoft.Health.Dicom.Functions/Microsoft.Health.Dicom.Functions.csproj:9]: why replace those assemblies with Framework Microsoft.AspNetCore.App?

File: Microsoft.Health.Dicom.sln (hunk 3 above)
Comment (wsugarman) [reply Microsoft.Health.Dicom.sln:44]: The project type will change on its own automatically and inadvertently in people's PR. I'm just getting out ahead of it lol

File: src/Microsoft.Health.Dicom.Functions/Microsoft.Health.Dicom.Functions.csproj (hunk 4 above)
Comment (wsugarman) [reply src/Microsoft.Health.Dicom.Functions/Microsoft.Health.Dicom.Functions.csproj:9]: Ultimately we're going to use ASP.NET Core components for our HTTP triggers. As such, we're essentially an ASP.NET Core project being hosted by someone else's ASP.NET Core project. So I think this framework reference makes it clearer this is an ASP.NET Core project and will automatically import all of the necessary dependencies.

**Review by yuedai0531 — APPROVED** [review]

File: Microsoft.Health.Dicom.sln (hunk 3 above)
Comment (pengchen0692) [reply Microsoft.Health.Dicom.sln:44]: lol

**Review by pengchen0692 — APPROVED** [review]

**Review by pengchen0692 — APPROVED** [review]

**Review by pengchen0692 — APPROVED** [review]

**Comment by wsugarman** [comment]
- Use latest 3.1.x .NET Core SDK for both SQL code-gen and Az Func tooling
- Use .NET 5.0 SDK for Az Func docker build
- Fix target framework specification
- Remove global.json for Az Func project

**Pull request merged**